doc/arkode/guide/source/ARKodeSPRKStorage.rst

```diff
@@ -0,0 +1,223 @@
+
+   Create a copy of the ARKodeSPRKStorage structure.
+
+   :param B: The original ARKodeSPRKStorage structure.
+   :return: Pointer to the copied ARKodeSPRKStorage structure.
+
+.. c:function:: void ARKodeSPRKStorage_Space(ARKodeSPRKStorage B, sunindextype* liw, sunindextype* lrw)
+
+   Get the workspace sizes required for the ARKodeSPRKStorage structure.
+
+   :param B: The ARKodeSPRKStorage structure.
+   :param liw: Pointer to store the integer workspace size.
+   :param lrw: Pointer to store the real workspace size.
+
+.. c:function:: void ARKodeSPRKStorage_Free(ARKodeSPRKStorage B)
+
+   Free the memory allocated for the ARKodeSPRKStorage structure.
+
+   :param B: The ARKodeSPRKStorage structure to free.
+
+.. c:function:: int ARKodeSPRKStorage_ToButcher(ARKodeSPRKStorage sprk_storage, ARKodeButcherTable* a_ptr, ARKodeButcherTable* b_ptr)
+
+   Convert the ARKodeSPRKStorage structure to the Butcher table representation.
+
+   :param sprk_storage: The ARKodeSPRKStorage structure.
+   :param a_ptr: Pointer to store the explicit Butcher table.
+   :param b_ptr: Pointer to store the diagonally-implicit Butcher table.
+
+.. c:function:: ARKodeSPRKStorage ARKodeSymplecticEuler()
+
+   Create the ARKodeSPRKStorage structure for the Symplectic Euler method.
+
+.. c:function:: ARKodeSPRKStorage ARKodeSymplecticLeapfrog2()
+
+   Create the ARKodeSPRKStorage structure for the Symplectic Leapfrog 2-2 method.
+
+.. c:function:: ARKodeSPRKStorage ARKodeSymplecticPseudoLeapfrog2()
+
+   Create the ARKodeSPRKStorage structure for the Symplectic Pseudo Leapfrog 2-2 method.
+
+.. c:function:: ARKodeSPRKStorage ARKodeSymplecticRuth3()
+
+   Create the ARKodeSPRKStorage structure for the Symplectic Ruth 3-3 method.
+
+.. c:function:: ARKodeSPRKStorage ARKodeSymplecticCandyRozmus4()
+
+   Create the ARKodeSPRKStorage structure for the Symplectic Candy Rozmus 4-4 method.
+
+.. c:function:: ARKodeSPRKStorage ARKodeSymplecticMcLachlan2()
+
+   Create the ARKodeSPRKStorage structure for the Symplectic McLachlan 2-2 method.
+
+.. c:function:: ARKodeSPRKStorage ARKodeSymplecticMcLachlan3()
+
+   Create the ARKodeSPRKStorage structure for the Symplectic McLachlan 3-3 method.
+
+.. c:function:: ARKodeSPRKStorage ARKodeSymplecticMcLachlan4()
+
+   Create the ARKodeSPRKStorage structure for the Symplectic McLachlan 4-4 method.
+
+.. c:function:: ARKodeSPRKStorage ARKodeSymplecticMcLachlan5()
+
+   Create the ARKodeSPRKStorage structure for the Symplectic McLachlan 5-6 method.
+
+.. c:function:: ARKodeSPRKStorage ARKodeSymplecticYoshida6()
+
+   Create the ARKodeSPRKStorage structure for the Symplectic Yoshida 6-8 method.
+
+.. c:function:: ARKodeSPRKStorage ARKodeSymplecticMcLachlan8()
+
+   Create the ARKodeSPRKStorage structure for the Symplectic McLachlan 8-16 method.
+
+.. c:function:: ARKodeSPRKStorage ARKodeSymplecticSofroniou10()
+
+   Create the ARKodeSPRKStorage structure for the Symplectic Sofroniou 10-36 method.
```

Comment:
I think we can remove them from the API, but they are useful internally to avoid duplicate code. The alternative is to use the macro approach we use for the Butcher tables, but I opted not to do that here since there are not nearly as many methods.